plugins/woocommerce/client/blocks/assets/js/blocks/product-elements/add-to-cart-form/frontend.ts

```diff
@@ -32,34 +39,118 @@ const getInputData = ( event: HTMLElementEvent< HTMLButtonElement > ) => {
+		? parseInt( childProductIdMatch[ 1 ], 10 )
+		: undefined;
 	return {
 		currentValue,
 		minValue,
 		maxValue,
 		step,
 		inputElement,
+		childProductId,
 	};
 };
 
 const dispatchChangeEvent = ( inputElement: HTMLInputElement ) => {
 	const event = new Event( 'change' );
-
 	inputElement.dispatchEvent( event );
 };
 
 store( 'woocommerce/add-to-cart-form', {
-	state: {},
+	state: {
+		get allowsDecrease() {
+			const context = getContext< Context >();
+			const { quantity, childProductId, productType } = context;
+			let currentQuantity = 0;
+			let selector =
+				'.wc-block-components-quantity-selector__input[name="quantity"]';
+			if ( productType === 'grouped' && childProductId ) {
+				currentQuantity = quantity?.[ childProductId ] || 0;
+				selector = `.wc-block-components-quantity-selector__input[name="quantity[${ childProductId }]"]`;
+			} else {
+				currentQuantity =
+					quantity?.[ context.productId as number ] || 0;
+			}
+			const inputElement = document.querySelector(
+				selector
+			) as HTMLInputElement | null;
+
+			if ( ! inputElement ) {
+				return false;
+			}
+
+			const parsedMinValue = parseInt( inputElement.min, 10 );
+			const parsedStep = parseInt( inputElement.step, 10 );
+			const defaultMinValue = childProductId ? 0 : 1;
+			const minValue = isNaN( parsedMinValue )
+				? defaultMinValue
+				: parsedMinValue;
+
+			const step = isNaN( parsedStep ) ? 1 : parsedStep;
+			return currentQuantity - step >= minValue;
+		},
+		get allowsIncrease() {
+			const context = getContext< Context >();
+			const { quantity, childProductId, productType } = context;
+			let currentQuantity = 0;
+			let selector =
+				'.wc-block-components-quantity-selector__input[name="quantity"]';
+			if ( productType === 'grouped' && childProductId ) {
+				currentQuantity = quantity?.[ childProductId ] || 0;
+				selector = `.wc-block-components-quantity-selector__input[name="quantity[${ childProductId }]"]`;
+			} else {
+				currentQuantity =
+					quantity?.[ context.productId as number ] || 0;
+			}
+			const inputElement = document.querySelector(
+				selector
+			) as HTMLInputElement | null;
+
+			if ( ! inputElement ) {
+				return false;
+			}
+
+			const parsedMaxValue = parseInt( inputElement.max, 10 );
+			const parsedStep = parseInt( inputElement.step, 10 );
+
+			const maxValue = isNaN( parsedMaxValue )
+				? undefined
+				: parsedMaxValue;
+			const step = isNaN( parsedStep ) ? 1 : parsedStep;
+			return maxValue === undefined || currentQuantity + step <= maxValue;
+		},
+	},
 	actions: {
 		addQuantity: ( event: HTMLElementEvent< HTMLButtonElement > ) => {
 			const inputData = getInputData( event );
 			if ( ! inputData ) {
 				return;
 			}
-			const { currentValue, maxValue, step, inputElement } = inputData;
+			const context = getContext< Context >();
+			const {
+				currentValue,
+				maxValue,
+				step,
+				inputElement,
+				childProductId,
+			} = inputData;
 			const newValue = currentValue + step;
 
```

Comment:
I wonder if we should save a variable like this at the beginning of the functions:

```TS
			const productId = childProductId || context.productId;
			if ( ! productId ) {
				return false;
			} 
```

If you do so, you will be able to avoid the `if { ... } else { ... }` and won't have to add `as number` every time you use `context.productId`.